cryptoservice/crypto_service_test.go

```diff
@@ -300,6 +300,40 @@ func (c CryptoServiceTester) TestListFromMultipleKeystores(t *testing.T) {
 	}
 }
 
+// asserts that adding a key adds to all keystores
+// and adding an existing key either succeeds if the role matches or fails if it does not
+func (c CryptoServiceTester) TestAddKey(t *testing.T) {
+	cryptoService := c.cryptoServiceFactory()
+	cryptoService.keyStores = append(cryptoService.keyStores,
+		trustmanager.NewKeyMemoryStore(passphraseRetriever))
+
+	privKey, err := trustmanager.GenerateECDSAKey(rand.Reader)
+	assert.NoError(t, err)
+
+	// Add the key to the targets role
+	assert.NoError(t, cryptoService.AddKey(privKey, data.CanonicalTargetsRole))
+
+	// Check that we added the key and its info to both keystores
```

Comment:
You're right -- `GetKey`/`GetPrivateKey` should work as long as the key is stored in just one store.  After thinking about this, I don't think there is a use case so I'll change behavior to be consistent with `Create`